src/Microsoft.IdentityModel.Tokens/JsonWebKeySet.cs

```diff
@@ -123,68 +123,100 @@ public JsonWebKeySet(string json, JsonSerializerSettings jsonSerializerSettings)
         /// </summary>
         public IList<SecurityKey> GetSigningKeys()
         {
-            List<SecurityKey> keys = new List<SecurityKey>();
-            for (int i = 0; i < Keys.Count; i++)
-            {
-                JsonWebKey webKey = Keys[i];
+            var keys = new List<SecurityKey>();
 
-                if (!StringComparer.Ordinal.Equals(webKey.Kty, JsonWebAlgorithmsKeyTypes.RSA))
+            foreach (var webKey in Keys)
+            {
+                // skip if "use" (Public Key Use) parameter is not empty or "sig"
+                if (!(string.IsNullOrWhiteSpace(webKey.Use) || webKey.Use.Equals(JsonWebKeyUseNames.Sig, StringComparison.Ordinal)))
+                {
+                    LogHelper.LogInformation(LogHelper.FormatInvariant(LogMessages.IDX10808, webKey.Use));
                     continue;
+                }
```

Comment:
great optimization, is this relevant for anything which is required to be const speed?

Reply:
No, we are just filtering out signing keys from the metadata received from a discovery endpoint.
There might be some keys used for encryption ("use": "enc") or some not defined by the standard, and we do not want these.

Reply:
Can we use StringComparison.OrdinalCaseInsensitive?

Reply:
We can relax this comparison but the parameter values are case-sensitive [by the standard](https://tools.ietf.org/html/rfc7517#section-4.2).